VideoApp/VideoApp/Stores/MainParticipant/MainParticipantStore.swift

```diff
@@ -17,7 +17,26 @@
 import Foundation
 
 class MainParticipantStore {
-    private(set) var mainParticipant: Participant
+    private(set) var mainParticipant: Participant {
+        didSet {
+            videoTrack = mainParticipant.mainVideoTrack
+
+            if mainParticipant.isPinned || videoTrack === mainParticipant.screenTrack || !mainParticipant.isDominantSpeaker {
+                videoTrack?.priority = .high
```

Comment:
Any participant can be pinned including local. And if there are no remote participants the local participant is displayed in the main view. 

In order to prevent problems like you mention (and keep this code pretty simple) I had to create a VideoTrack protocol similar to my Participant protocol. And then I have a LocalVideoTrack concrete and RemoteVideoTrack concrete very similar to what I did for participants previously. LocalVideoTrack does nothing when `priority` is set so totally safe. And of course RemoteVideoTrack calls down into TwilioVideo.RemoteVideoTrack when `priority` is set.

Reply:
Yeah, I saw that later on as I scrolled. 👍